File: jewels_and_stones.py
class Solution(object):
    def numJewelsInStones(self, jewels, stones):
        hashmap = {}
        count = 0
        for i, x in enumerate(jewels):
            hashmap[x[i]] = True
        
        for i, y in enumerate(stones):
            if y in hashmap:
                count += 1
        return count
    
# O(n)
#O(n)

File: jewelsAndStones.js
var numJewelsInStones = function(jewels, stones){
    let obj = {};
    let count = 0;
    for(const jewel in jewels){
        obj[jewel] = true
    }

    for(const stone of stones){
        if(stone in obj){
            count++
        }
    }
    return count
}